File: API_Server/Taxi.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Taxi.Domain.Entities
{
    public class Order
    {
        public int OrderId { get; set; }
        public string PickupLocation { get; set; }
        public string ReturnLocation { get; set; }
        //status == 1 => order have been done, opposite
        public bool Status { get; set; }
        public int Price { get; set; }
        //Number of seats reserved
        public int Reservations { get; set; }
        public bool IsCancel { get; set; }
        public int Rate { get; set; }
        public string RateContent { get; set; }
        public DateTime Created { get; set; }
        public int DriverId { get; set; }
        public virtual Driver Driver { get; set; }
        public int CustomerId { get; set; }
        public virtual Customer Customer { get; set; }
    }
}

File: Client/src/app/shared/models/order.ts
import { Adapter } from './../services/Adapter';
import { Driver } from './driver';
import { Customer } from './customer';
import { Injectable } from '@angular/core';
export class Order {
  constructor(
    public orderId: number,
    public pickUpLocation: string,
    public returnLocation: string,
    public status: boolean,
    public isCancel: boolean,
    public price: number,
    public rate: number,
    public rateContent: string,
    public created: Date,
    public driverId: number,
    public driverName: string,
    public driverImg: string,
    public driverRate: number,
    public customerId: string,
    public customerPhone: string,
    public customerName: string,
    public customerImg: string
  ) {}
 
}

@Injectable({
  providedIn: 'root',
})
export class OrderAdapter implements Adapter<Order> {
  adapt(item: any): Order {
    return new Order(
      item.orderId,
      item.pickupLocation,
      item.returnLocation,
      item.status,
      item.isCancel,
      item.price,
      item.rate,
      item.rateContent,
      item.created,
      item.driverId,
      item.driverName,
      item.driverImg,
      item.driverRate,
      item.customerId,
      item.customerPhone,
      item.customerName,
      item.customerImg
    );
  }
}
